sw/device/silicon_creator/rom/rom.c

```diff
@@ -455,13 +456,17 @@ void rom_main(void) {
   shutdown_finalize(rom_try_boot());
 }
 
-void rom_interrupt_handler(void) { shutdown_finalize(rom_irq_error()); }
+void rom_interrupt_handler(void) {
+  register rom_error_t error asm("a0") = rom_irq_error();
+  asm volatile("tail shutdown_finalize;" ::"r"(error));
```

Comment:
Isn't `shutdown_finalize` already `noreturn`?  The compiler still emits a `jal <target>` for a `noreturn` function?

Reply:
Correct, `shutdown_finalize()` is already `noreturn`. Even removing `interrupt` and marking the handlers `noreturn` didn't help. I wonder if it's because it's in a different TU.

@luismarques any thoughts?

Reply:
It's a missing optimization for RISC-V. I'll submit an LLVM patch to address that, but we should use the inline assembly for now.

Reply:
Thanks for the clarification @luismarques! I'll go ahead and merge this now.